addons/crm/tests/test_crm_lead.py

```diff
@@ -49,25 +49,66 @@ def test_crm_lead_creation_no_partner(self):
         lead.partner_id = False
         self.assertEqual(lead.country_id, belgium, "Country should still be set on the lead")
 
+    def test_crm_lead_creation_partner_address(self):
+        belgium = self.env.ref('base.be')
+        empty_partner = self.env['res.partner'].create({
+            'name': 'Empty partner',
+            'country_id': belgium.id,
+        })
+        lead_data = {
+            'name': 'Test',
+            'street': 'My street',
+            'street2': 'My street',
+            'city': 'My city',
+            'zip': 'test@odoo.com',
+            'state_id': self.env['res.country.state'].create({
+                'name': 'My state',
+                'country_id': belgium.id,
+                'code': 'MST',
+            }).id,
+            'country_id': belgium.id,
+        }
+        lead = self.env['crm.lead'].create(lead_data)
+        lead.partner_id = empty_partner
+        self.assertEqual(lead.street, empty_partner.street, "Street should be sync from the Partner")
+        self.assertEqual(lead.street2, empty_partner.street2, "Street 2 should be sync from the Partner")
+        self.assertEqual(lead.city, empty_partner.city, "City should be sync from the Partner")
+        self.assertEqual(lead.zip, empty_partner.zip, "Zip should be sync from the Partner")
+        self.assertEqual(lead.state_id, empty_partner.state_id, "State should be sync from the Partner")
+        self.assertEqual(lead.country_id, empty_partner.country_id, "Country should be sync from the Partner")
```

Comment:
You wouldn't be able to tell the difference, since both records start with the same country. Anyway not sure we need to cover *every* address field here ;-)